test_onedim.TestFreeFlame fails on OSX 10.15.6

<!-- Please fill in the following information to report a problem with Cantera. If you have a question about using Cantera, please post it on our Google Users' Group (https://groups.google.com/forum/#!forum/cantera-users). Feature enhancements should be discussed in the dedicated Cantera enhancements repository (https://github.com/Cantera/enhancements/new/choose) -->

**Problem description**
test_onedim.TestFreeFlame fails on OSX

<!-- A clear and concise description of what the bug is. -->

**Steps to reproduce**
Run `scons test `

<!-- A minimal, complete, and reproducible example demonstrating the problem. See https://stackoverflow.com/help/minimal-reproducible-example for additional suggestions on how to create such an example. -->


**Behavior**
======================================================================
FAIL: test_write_hdf (cantera.test.test_onedim.TestFreeFlame)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/santosh1/packages/canteraSource/build/python/cantera/test/test_onedim.py", line 649, in test_write_hdf
    meta = f.read_hdf(filename)
  File "/Users/santosh1/packages/canteraSource/build/python/cantera/onedim.py", line 635, in read_hdf
    self.settings = meta
  File "/Users/santosh1/packages/canteraSource/build/python/cantera/onedim.py", line 663, in settings
    setattr(self, key, s[key])
  File "interfaces/cython/cantera/onedim.pyx", line 1411, in cantera._cantera.Sim1D.fixed_temperature.__set__
cantera._cantera.CanteraError: 
***********************************************************************
CanteraError thrown by StFlow::setupGrid:
grid points must be monotonically increasing
***********************************************************************

<!-- Describe the result of executing the above steps, and how this differs from what you expect to happen. -->

**System information**

- Cantera version: [2.5.0b1 commit # 2cd06194e]
- OS: OSX 10.15.6 Catalina
- Python/MATLAB/other software versions: XCode 11.7, python 3.7, numpy 1.17

**Attachments**

<!-- If applicable, attach scripts and/or input files to help explain your problem. Please do *not* attach screenshots of code or terminal output. -->

**Additional context**
@bryanwweber could reproduce these, but the CI could not 
<!-- Add any other context about the problem here. -->

@ischoegl Here is the log of the second failure:

======================================================================
FAIL: test_ion_profile (cantera.test.test_onedim.TestIonBurnerFlame)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/santosh1/packages/canteraSource/build/python/cantera/test/test_onedim.py", line 1253, in test_ion_profile
    self.assertNear(max(self.sim.E), 552.33, 1e-2)
  File "/Users/santosh1/packages/canteraSource/build/python/cantera/test/utilities.py", line 57, in assertNear
    self.fail(message)
AssertionError: AssertNear: 580.09163013629 - 552.33 = 27.761630136288
Relative error of 4.903055e-02 exceeds rtol = 1.000000e-02

----------------------------------------------------------------------

--- a/src/oneD/Sim1D.cpp
+++ b/src/oneD/Sim1D.cpp
@@ -435,7 +435,7 @@ int Sim1D::setFixedTemperature(double t)
     std::vector<size_t> dsize;
 
     for (size_t n = 0; n < nDomains(); n++) {
-        bool addnewpt=false;
+        bool addnewpt = false;
         Domain1D& d = domain(n);
         size_t comp = d.nComponents();
 
@@ -445,14 +445,21 @@ int Sim1D::setFixedTemperature(double t)
         size_t npnow = d.nPoints();
         size_t nstart = znew.size();
         if (d_free && d_free->domainType() == cFreeFlow) {
+            double ttol = 1.e-3; // avoid new point too close to existing point
             for (size_t m = 0; m < npnow-1; m++) {
-                if (value(n,2,m) == t) {
+                if (fabs(t - value(n, 2, m)) <= ttol * t) {
                     zfixed = d.grid(m);
                     d_free->m_zfixed = zfixed;
                     d_free->m_tfixed = t;
                     addnewpt = false;
                     break;
-                } else if ((value(n,2,m)<t) && (value(n,2,m+1)>t)) {
+                } else if (fabs(value(n, 2, m + 1) - t) <= ttol * t) {
+                    zfixed = d.grid(m + 1);
+                    d_free->m_zfixed = zfixed;
+                    d_free->m_tfixed = t;
+                    addnewpt = false;
+                    break;
+                } else if ((value(n, 2, m) - t) * (value(n, 2, m + 1) - t) < 0.) {
                     z1 = d.grid(m);
                     m1 = m;
                     z2 = d.grid(m+1);

--- a/src/oneD/Sim1D.cpp
+++ b/src/oneD/Sim1D.cpp
@@ -429,53 +429,49 @@ int Sim1D::setFixedTemperature(double t)
 {
     int np = 0;
     vector_fp znew, xnew;
-    doublereal zfixed = 0.0;
-    doublereal z1 = 0.0, z2 = 0.0, t1,t2;
-    size_t m1 = 0;
+    double zfixed = 0.0;
+    double z1 = 0.0, z2 = 0.0;
     std::vector<size_t> dsize;
 
     for (size_t n = 0; n < nDomains(); n++) {
-        bool addnewpt = false;
         Domain1D& d = domain(n);
         size_t comp = d.nComponents();
+        size_t mfixed = npos;
 
-        // loop over points in the current grid to determine where new point is
-        // needed.
+        // loop over current grid to determine where new point is needed
         StFlow* d_free = dynamic_cast<StFlow*>(&domain(n));
         size_t npnow = d.nPoints();
         size_t nstart = znew.size();
         if (d_free && d_free->domainType() == cFreeFlow) {
-            double ttol = 1.e-3; // avoid new point too close to existing point
-            for (size_t m = 0; m < npnow-1; m++) {
-                if (fabs(t - value(n, 2, m)) <= ttol * t) {
-                    zfixed = d.grid(m);
-                    d_free->m_zfixed = zfixed;
-                    d_free->m_tfixed = t;
-                    addnewpt = false;
-                    break;
-                } else if (fabs(value(n, 2, m + 1) - t) <= ttol * t) {
-                    zfixed = d.grid(m + 1);
-                    d_free->m_zfixed = zfixed;
-                    d_free->m_tfixed = t;
-                    addnewpt = false;
-                    break;
-                } else if ((value(n, 2, m) - t) * (value(n, 2, m + 1) - t) < 0.) {
-                    z1 = d.grid(m);
-                    m1 = m;
-                    z2 = d.grid(m+1);
-                    t1 = value(n,2,m);
-                    t2 = value(n,2,m+1);
+            for (size_t m = 0; m < npnow - 1; m++) {
+                bool fixedpt = false;
+                double t1 = value(n, 2, m);
+                double t2 = value(n, 2, m + 1);
+                // threshold to avoid adding new point too close to existing point
+                double thresh = min(1., 1.e-1 * (t2 - t1));
+                z1 = d.grid(m);
+                z2 = d.grid(m + 1);
+                if (fabs(t - t1) <= thresh) {
+                    zfixed = z1;
+                    fixedpt = true;
+                } else if (fabs(t2 - t) <= thresh) {
+                    zfixed = z2;
+                    fixedpt = true;
+                } else if ((t1 < t) && (t < t2)) {
+                    mfixed = m;
+                    zfixed = (z1 - z2) / (t1 - t2) * (t - t2) + z2;
+                    fixedpt = true;
+                }
 
-                    zfixed = (z1-z2)/(t1-t2)*(t-t2)+z2;
+                if (fixedpt) {
                     d_free->m_zfixed = zfixed;
                     d_free->m_tfixed = t;
-                    addnewpt = true;
                     break;
-                    //copy solution domain and push back values
                 }
             }
         }
 
+        // copy solution domain and push back values
         for (size_t m = 0; m < npnow; m++) {
             // add the current grid point to the new grid
             znew.push_back(d.grid(m));
@@ -484,11 +480,11 @@ int Sim1D::setFixedTemperature(double t)
             for (size_t i = 0; i < comp; i++) {
                 xnew.push_back(value(n, i, m));
             }
-            if (m==m1 && addnewpt) {
-                //add new point at zfixed
+            if (m == mfixed) {
+                // add new point at zfixed (mfixed is not npos)
                 znew.push_back(zfixed);
                 np++;
-                double interp_factor = (zfixed-z2) / (z1-z2);
+                double interp_factor = (zfixed - z2) / (z1 - z2);
                 // for each component, linearly interpolate
                 // the solution to this point
                 for (size_t i = 0; i < comp; i++) {

